src/core/bitop.d

```diff
@@ -706,3 +706,38 @@ version (D_InlineAsm_X86_64)
         }
     }
 }
+
+/**
+ *  Bitwise rotates unsigned arithmetic types
+ */
+pure T rotl(T)(in T u, in uint s)
+    if (__traits(isArithmetic, T) && __traits(isUnsigned, T))
```

Comment:
Bit operations on floats are only possible after reinterpreting the representation to an integer, so please `isIntegral` here.